### PR TITLE
Add continuation flag and update README

### PR DESCRIPTION
**Pull request by Roman4466**
This pull request introduces the following changes:

1. Modified 1_download_dialogs_data.py:
   - Added a new `--continuation` flag to allow resuming downloads from the last downloaded message and download in batches of 10,000 messages
   - Added the `download_dialog_within_batches` function to support the continuation mode.

2. Updated the README.md

Please review these changes and let me know if any further modifications are needed.

### CODE REVIEW

**Review by SanGreel — CHANGES_REQUESTED** [review]
1. please update your branch from the master
2. use black (https://pypi.org/project/black/) util to format the code, either use format from the current master branch.
3

File: 1_download_dialogs_data.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def init_args():
     parser.add_argument('--skip_private', action='store_true')
     parser.add_argument('--skip_groups', action='store_true')
     parser.add_argument('--skip_channels', action='store_true')
+    parser.add_argument('--continuation', action='store_true', help="Enable continuation mode to resume from last downloaded message")
```
Comment (SanGreel) [inline 1_download_dialogs_data.py:48]: continuation ->resume_from_last_batch
add type and default value

File: 1_download_dialogs_data.py [hunk 2]
```diff
@@ -228,6 +228,84 @@ async def download_dialog(client, id, MSG_LIMIT, config):
...
+    all_messages = []
+    offset_id = None
+    batch_size = 10000  # Download 10,000 messages at a time
+    total_messages = 0
+
+    if os.path.exists(dialog_file_path) and os.path.getsize(dialog_file_path) > 0:
+        print(f"Continuation mode: Found existing CSV for dialog #{id}. Resuming download...")
+        df = pd.read_csv(dialog_file_path)
+        if not df.empty:
+            last_row = df.iloc[-1]
```
Comment (SanGreel) [inline 1_download_dialogs_data.py:256]: it can be not the most recent msg, it should be the latest msg

File: 1_download_dialogs_data.py [hunk 3]
```diff
@@ -262,4 +341,7 @@ async def download_dialog(client, id, MSG_LIMIT, config):
         print(f"Loading dialog #{id}")
 
         with client:
-            client.loop.run_until_complete(download_dialog(client, id, MSG_LIMIT, config))
+            if(CONTINUATION):
```
Comment (SanGreel) [inline 1_download_dialogs_data.py:344]: Logic in this block doesn't work
download_dialog - expects 4 args, you give 5

did you try to run your code?

please double check it works,
remove "CONTINUATION" variable

logic should be either do it all in one or go by batches

default logic should be to go by batches with restore from the last step

File: README.md [hunk 4]
```diff
@@ -18,7 +18,8 @@ path to config file
 `--debug_mode`
 Debug mode
 
-
+`--continuation`
+Before loading dialog checks whether it is in your `../data/dialogs` folder, starts loading messages from last item in csv if `.csv` with that id exist. Downloads messages in batches of 10,000 messages.
```
Comment (SanGreel) [inline README.md:22]: Please rewrite this with updated logic and fix mistakes.

**Review by SanGreel — CHANGES_REQUESTED** [review]

File: 1_download_dialogs_data.py (hunk 3 above)
Comment (Roman4466) [reply 1_download_dialogs_data.py]: I believe I fixed it in a better way than you suggested. Please take a look on my update and leave feedback

**Pull request merged**